amqp/src/main/scala/akka/stream/alpakka/amqp/AmqpConnectorSettings.scala

```diff
@@ -153,22 +154,52 @@ object TemporaryQueueSourceSettings {
     TemporaryQueueSourceSettings(connectionProvider, exchange)
 }
 
+final class AmqpPublishConfirmConfiguration private (
+    val confirmTimeout: FiniteDuration
+) {
+  override def toString: String =
+    "AmqpPublishConfirmConfiguration(" +
+    s"confirmTimeout=$confirmTimeout" +
+    ")"
+}
+
+object AmqpPublishConfirmConfiguration {
+  def apply(confirmTimeout: FiniteDuration): AmqpPublishConfirmConfiguration =
+    new AmqpPublishConfirmConfiguration(confirmTimeout)
+
+  def unapply(settings: AmqpPublishConfirmConfiguration): Option[FiniteDuration] =
```

Comment:
Extractors are hard to evolve in a binary compatible way, please remove and access the value directly where needed.